autogen/oai/gemini.py

```diff
@@ -68,14 +72,33 @@ class GeminiClient:
         "max_output_tokens": "max_output_tokens",
     }
 
+    def initialize_vartexai(self, **params):
+        if "google_application_credentials" in params:
+            # Path to JSON Keyfile
+            os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = params["google_application_credentials"]
+        vertexai_init_args = {}
+        if "project_id" in params:
+            vertexai_init_args["project"] = params["project_id"]
+        if "location" in params:
+            vertexai_init_args["location"] = params["location"]
+        if vertexai_init_args:
+            vertexai.init(**vertexai_init_args)
+
     def __init__(self, **kwargs):
         self.api_key = kwargs.get("api_key", None)
         if not self.api_key:
             self.api_key = os.getenv("GOOGLE_API_KEY")
-
-        assert (
-            self.api_key
-        ), "Please provide api_key in your config list entry for Gemini or set the GOOGLE_API_KEY env variable."
+            if self.api_key is None:
```

Comment:
The docstring is missing. For new comers to this Gemini feature, they may have a hard time to understand the logic here. Give a docstring to tell users there are 2 different ways to initialize the Gemini client, and how to use these 2 ways with "llm_config".

Reply:
Now there is a docstring with detailed information about the authentication methods